zerver/lib/narrow.py

```diff
@@ -32,6 +32,18 @@ def is_web_public_compatible(narrow: Iterable[Dict[str, str]]) -> bool:
             return False
     return True
 
+def is_web_public_narrow(narrow: Optional[Iterable[Dict[str, Any]]]) -> bool:
+    if narrow is None:
+        return False
+
+    for term in narrow:
+        # Web public queries are only allowed for limited types of narrows.
+        # term == {'operator': 'streams', 'operand': 'web-public', 'negated': False}
+        if term['operator'] == 'streams' and term['operand'] == 'web-public' and term['negated'] is False:
```

Comment:
If `{'operator': 'streams', 'operand': 'web-public', 'negated': False}` is exactly the expected form of `term`, shouldn't we just compare the dictionaries rather than values under specific keys? It might not matter right now, but if these `term` dicts were ever to acquire an additional key, it wouldn't play safely with this `if`

Reply:
The intent is to add additional allowed operators in the future.

(And I think we don't have plans to change that format, and if we would, we'd need to rewrite this whole file)